inc/targetbase.class.php

```diff
@@ -1049,112 +1049,124 @@ protected function parseTags($content, PluginFormcreatorFormAnswer $formanswer,
+
+         // Search for placeholders
+         $matches = [];
+         $regex = "/##answer_$questionID\.(?<property>[a-zA-Z0-9_.]+)##/";
+         preg_match_all($regex, $content, $matches);
+
+         // For each placeholder found
+         foreach ($matches["property"] as $property) {
+            $placeholder = "##answer_$questionID.$property##";
+            // Convert Property_Name to Property Name
+            $property = str_replace("_", " ", $property);
+            $searchOption = $item->getSearchOptionByField("name", $property);
+
+            // Execute search
+            $data = Search::prepareDatasForSearch(get_class($item), [
+               'criteria' => [
+                  [
+                     'field'      => $searchOption['id'],
+                     'searchtype' => "contains",
+                     'value'      => "",
+                  ],
+                  [
+                     'field'      => 2,
+                     'searchtype' => "equals",
+                     'value'      => $answer,
                   ]
-               ]);
-               Search::constructSQL($data);
-               Search::constructData($data);
-
-               $propertyValue = "";
-               foreach ($data['data']['rows'] as $row) {
-                  $targetKey = get_class($item) . "_" . $searchOption['id'];
-                  // Add each result
-                  for ($i=0; $i < $row[$targetKey]['count']; $i++) {
-                     $propertyValue .=$row[$targetKey][$i]['name'];
-                     if ($i+1 < $row[$targetKey]['count']) {
-                        $propertyValue .= ", ";
-                     }
+               ]
+            ]);
+            Search::constructSQL($data);
+            Search::constructData($data);
+
+            // Handle search result, there may be multiple values
+            $propertyValue = "";
+            foreach ($data['data']['rows'] as $row) {
+               $targetKey = get_class($item) . "_" . $searchOption['id'];
+               // Add each result
+               for ($i=0; $i < $row[$targetKey]['count']; $i++) {
+                  $propertyValue .=$row[$targetKey][$i]['name'];
+                  if ($i+1 < $row[$targetKey]['count']) {
+                     $propertyValue .= ", ";
                   }
                }
-
-               // Replace placeholder in content
-               $content = str_replace(
-                  $placeholder,
-                  Toolbox::addslashes_deep($propertyValue),
-                  $content
-               );
             }
+
+            // Replace placeholder in content
+            $content = str_replace(
+               $placeholder,
+               Toolbox::addslashes_deep($propertyValue),
+               $content
+            );
+
+            return $content;
          }
+      } finally {
+         // Put the old locales on succes or if an expection was thrown
+         $TRANSLATE->setLocale($oldLocale);
       }
-
-      return $content;
```

Comment:
if we catch an exception, the method will not return anything. Either we need to throw an exception (and catch it in the caller), either we should return a value.